### ISSUE TITLE
Kite keeps coming back

### ISSUE THREAD

**Issue by lUNuXl** [kiteco/issue-tracker#524]
**Required Information**
To help us triage your issue effectively, please make sure you've completed these steps before opening this issue. Issues with not enough information may be deprioritized or ignored.

- [Not applicable ] [Read our guide](https://help.kite.com/article/96-contact-us-with-logs) on how to report an issue.
- [Not applicable ] Include your Kite application logs, _taken while you are experiencing your issue_.
- [ Not applicable] If reporting a performance problem, include your resource usage logs, _taken while you are experiencing your issue_.
- [ Probably?] If reporting an issue with Kite's editor plugin, include which editor you are using and the version, and any accompanying logs from your editor's console.

**Describe the bug**
A long time ago I was trying to make a python IDE out of sublime text 3, so I installed a few extensions (kite among the others) and I think this package: https://aur.archlinux.org/packages/kite/ , then after deciding that this wasn't working for me I just uninstalled the plugins and etc.

The "official" way of removing kite from sublime wasn't working as far as I remember so I just removed its directory from sublime plugins, I remember that it it then managed to come back somehow I think (I'm not sure I didn't install the package again to test something).

Now, about a month later since that sublime experiment, sometime this/last week I noticed that I had a notification from kite in Intellij IDEA... and Kite plugin was automagically installed in that IDE... so I removed it, restarted the IDE, and then after Kite was gone from intellij I checked if kite was still on my machine with:

```
find -name kite
```
And it was, not only did it install itself for my IDE, it also installed itself for atom and vim which I don't use at all, so I deleted all of its directories and thought that it'll be gone for good.
But I was wrong.

It's back, and it installed itself **again** into IDEA (I just removed it) and atom... and vim.

I just wanted to make sure there wasn't any systemd service and/or package in /opt but it doesn't look like it's the case.
```
toaster@root / # find -name kite
find: './run/user/1000/gvfs': Permission denied
find: './tmp/.mount_jetbraG4c1Fz': Permission denied
./home/larry/tmp/starsector/data/variants/kite
./home/larry/.atom/packages/kite
./home/larry/.vim/pack/kite
./home/larry/.vim/pack/kite/start/vim-plugin/autoload/kite
```

How do I get rid of kite?

EDIT:

Now I opened sublime text and I was greeted with a message that kite is missing dependencies... after clicking cancel a few times the popup stopped appearing and tried removing kite plugin with package control (Kite Python Autocomplete).

And now my sublime text editor is frozen (the same thing that led me to manually removing the package's files)

**Screenshots**
Not applicable

**Comment by adamsmith** [comment]
It sounds like you have the Kite desktop app installed and running. Can you look for a process named `kited`?

See also https://help.kite.com/article/68-how-to-uninstall-kite

**Comment by adamsmith** [comment]
Please re-open if you're still running into issues. Thanks!

**Comment by lUNuXl** [comment]
> It sounds like you have the Kite desktop app installed and running. Can you look for a process named `kited`?
> 
> See also https://help.kite.com/article/68-how-to-uninstall-kite

The only kite system application I've ever installed was form AUR or official repositories, and I removed it afterwards.

There is no such process as kited: 
```
$ ps -aux | grep -i kited
larry     255492  0.0  0.0   6380  2228 pts/8    S+   23:41   0:00 grep -i kited
```

**Comment by lUNuXl** [comment]
> Please re-open if you're still running into issues. Thanks!

@adamsmith Can you reopen this issue for me? I don't see any option to reopen it.

**Comment by adamsmith** [comment]
Did the uninstall instructions linked above help, or turn up anything interesting?

**Comment by lUNuXl** [comment]
> Did the uninstall instructions linked above help, or turn up anything interesting?

No, I don't have the uninstall binary/script in my system anymore because I just did `rm -rf` on every directory in my system that referenced kite.

Also as pointed out in: https://github.com/kiteco/vscode-plugin/issues/375

I have the following files in my `~/.config/systemd/user`

```
$ ls ~/.config/systemd/user -lah
total 12K
drwxr-xr-x 1 larry users 196 Dec 23 01:04 .
drwxr-xr-x 1 larry users   8 Sep 29  2019 ..
drwxr-xr-x 1 larry users  36 Dec 23 01:04 basic.target.wants
drwxr-xr-x 1 larry users  78 Dec 23 01:04 default.target.wants
-rw------- 1 larry users 205 Dec 23 01:04 kite-autostart.service
-rw------- 1 larry users 154 Dec 23 01:04 kite-updater.service
-rw-r--r-- 1 larry users 114 Jul 23  2019 kite-updater.timer
```

```
$ cat  ~/.config/systemd/user/kite-autostart.service 
[Unit]
Description=kited autostart on first login of a user

[Service]
Type=simple
StandardOutput=null
ExecStart=/home/larry/.local/share/kite/current/kited --system-boot

[Install]
WantedBy=default.target
```
But the daemon executable doesn't exist so it cannot be it.
```
$ stat /home/larry/.local/share/kite/current/kited
stat: cannot statx '/home/larry/.local/share/kite/current/kited': No such file or directory
```

I just checked right now if removing `~/.config/sublime-text-3/Packages/Kite Python Autocomplete` would fix anything... after restarting sublime text the directory is back with the information about missing dependencies.

**Comment by adamsmith** [comment]
Hi @lUNuXl, thanks for this info. Let me try to get an engineer from our end to take a look.

**Comment by adamsmith** [comment]
It looks like you installed Kite via brew, which put `kited` in a non-standard directory. Given that `kited` is now deleted, you should not see any Kite plugins being installed by Kite. 

Is it possible that Sublime's Package Control is reinstalling the Kite plugin? See step 3 from https://paulsalaets.com/posts/manually-removing-package-from-sublime-text-2/

**Comment by lUNuXl** [comment]
Hello,

It's been a long time since I've had time for looking into this issue, lots of software updates in the meantime to both the sublime text and its plugins.

Today I sat down to work on this and tried removing it again with package control, and this time it worked without hanging my editor, so the problem can be considered to be solved I guess.

It's most likely that this was more of a package control / sublime text bug than a kite one.



> Given that `kited` is now deleted, you should not see any Kite plugins being installed by Kite.
> 
> Is it possible that Sublime's Package Control is reinstalling the Kite plugin? See step 3 from https://paulsalaets.com/posts/manually-removing-package-from-sublime-text-2/

For future reference - this issue is about sublime text 3, the link provided talks about sublime text 2, so it's not relevant (most likely).

**Comment by Patitotective** [comment]
I'm using sublime text and i have the same problem: I installed `Kite Python AutoComplete` Sublime Package and when installed it says that i need to install a lot of stuff and my computer isn't able to use Kite so i `Package Control: Remove Package` and selected `Kite Python AutoComplete`, but Sublime Text crashes. I tried to delete the folder in `sublime-text-3/Packages` but it install itself again (and popup a window to install Kite).

I already had this problem and I solved it but i forgot how.
